### ISSUE TITLE
C# Files have no syntax highlighting, intellisense, etc.

### ISSUE THREAD

**Issue by EjPlatzer** [dotnet/vscode-csharp#4935]
## Issue Description ##
When I open .cs files, they are treated as Plain Text. The 'Autodetect' language mode leaves them as plain text, and trying to manually set the language mode is impossible because there's no `csharp` language mode to choose:
![image](https://user-images.githubusercontent.com/35319956/145235557-8e0166d5-4649-491d-8bdd-912c1da0b94f.png)

## Steps to Reproduce ##
1. Open VSCode
2. Install this extension
3. Open a `.cs` file

## Expected Behavior ##
The file is automatically detected to be csharp, and syntax highlighting, intellisense, etc. is enabled

## Actual Behavior ##
The file is treated as plain text, and no syntax highlighting is possible.

### OmniSharp log ###
<details>Starting OmniSharp server at 12/8/2021, 10:22:41 AM
    Target: /Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI

OmniSharp server started.
    Path: /Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/run
    PID: 82307

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 12.0.1 (x64)
[info]: OmniSharp.Services.DotNetCliService
        Checking the 'DOTNET_ROOT' environment variable to find a .NET SDK
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to /Volumes/SSD/home/dev/.local/share/dotnet/dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 17.0.0 - "/Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild/Current/Bin/MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 17.0.0 - "/Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild/Current/Bin
            CscToolPath = /Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /Volumes/SSD/Applications/code-portable-data/extensions/ms-dotnettools.csharp-1.23.17/.omnisharp/1.37.17/omnisharp/.msbuild
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI/GordonPublicAPI.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI' on host 81995.
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI/GordonPublicAPI.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI/GordonPublicAPI.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Volumes/SSD/home/dev/Developer/PublicAPI/GordonPublicAPI/GordonPublicAPI.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: GordonPublicAPI
</details>

### C# log ###
The C# Log is completely blank
<details>
</details>

## Environment information ##

**VSCode version**: 1.62.3
**C# Extension**: 1.23.17

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.0
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /Volumes/SSD/home/dev/.local/share/dotnet/sdk/6.0.100/

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  6.0.100 [/Volumes/SSD/home/dev/.local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [/Volumes/SSD/home/dev/.local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/Volumes/SSD/home/dev/.local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-dotnettools|1.23.17|;

</details>

**Comment by EjPlatzer** [comment]
I would just like to note that, after hearing about the new C# Dev Kit extension and the updates to this extension, I came back to try it again >1.5 years later and am still seeing the same issue, on a completely separate installation of VS Code.

**Comment by EjPlatzer** [comment]
I have discovered that, when I create a new VS Code profile from an empty profile, and then install this extension, everything appears to work as expected. That suggests that either my settings are incorrect (even then, as far as I can tell, none of my defaults have changed in any way that would affect C#), or somehow my default profile is corrupt. Regardless, the issue itself is now resolved, since I can successfully use syntax highlighting for C# in VS Code, so I am closing this issue.
